### PR TITLE
fix(jsdocs): solve all lint warnings in project

### PR DESCRIPTION
**Pull request by rinaok**
Waiting for https://github.com/Vonage/vivid-3/pull/110

### CODE REVIEW

**Review by yinonov — CHANGES_REQUESTED** [review]
please await #861 as this is a major change applying many files

**Review by yinonov — CHANGES_REQUESTED** [review]
some of the ommited decorators like accessors (e.g. `@public`) are TSDoc related. we might need to go over them all and fix due to #818

**Comment by rinaok** [comment]
> please await #861 as this is a major change applying many files

@yinonov Now after #861 is merged, can we merge this?

**Comment by yinonov** [comment]
> > please await #861 as this is a major change applying many files
> 
> @yinonov Now after #861 is merged, can we merge this?

it requires a very thorough review still. the pain point with this symptom relates to https://github.com/Vonage/vivid-3/pull/110#issuecomment-1006352237

the optimal solution is, if tsdoc-lint was to enforce comments, to remove jsdoc-lint and use the tsdoc-lint instead

**Comment by rinaok** [comment]
> > > please await #861 as this is a major change applying many files
> > 
> > 
> > @yinonov Now after #861 is merged, can we merge this?
> 
> it requires a very thorough review still. the pain point with this symptom relates to [#110 (comment)](https://github.com/Vonage/vivid-3/pull/110#issuecomment-1006352237)
> 
> the optimal solution is, if tsdoc-lint was to enforce comments, to remove jsdoc-lint and use the tsdoc-lint instead

@yinonov But you said that we should avoid at this point from tsdoc https://github.com/Vonage/vivid-3/pull/110#issuecomment-1006352237

**Comment by yinonov** [comment]
Because of lint technical limitation not enforcing necessary comments. But if u already went through and added all, we can reconsider tsdoc lint.
It will do more, now that api-extractor generate meta data

**Comment by yinonov** [comment]
let's try to revive #110

**Comment by rinaok** [comment]
> let's try to revive #110

@yinonov merged https://github.com/Vonage/vivid-3/pull/110

File: libs/components/src/lib/breadcrumb/ui.test.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@ import {
 
 const components = ['breadcrumb', 'breadcrumb-item'];
 
+/**
+ *
+ *
+ * @returns {*} - template
+ */
```
Comment (YonatanKra) [inline libs/components/src/lib/breadcrumb/ui.test.ts:14]:
```suggestion
/**
 *
 *
 * @returns (): Promise<void>;
 */
```

**Review by YonatanKra — COMMENTED** [review]
I think we can exempt test files (`.spec|test.ts`) from tsdoc linting rule.
What do you think @olaf-k ?

**Review by yinonov — DISMISSED** [review]
As #110 expected to now redfine eslint rules, it might outdate the changes of this PR.

what worries me is the fact we are not that acquainted with [TSDoc](https://tsdoc.org) and how to use it to its extent

the general idea to automate and improve our workflow is to have comments populating to READMEs and from there to docs website

for web components interfaces, TSDoc lacks some features, of where `custom-elements-manifests` come into play.

1st, let's make 1, single component as perfect POC, have its doc generated automatically.

going over lots of files to fix without a solid understanding the result will probably buy us more debts

**Comment by olaf-k** [comment]
> I think we can exempt test files (`.spec|test.ts`) from tsdoc linting rule. What do you think @olaf-k ?

my gut feeling would be "yes please!" or at least keep only rules that prevent readability issues (weird capitalization, mix tab/space, this sort of stuff)

**Comment by yinonov** [comment]
> > I think we can exempt test files (`.spec|test.ts`) from tsdoc linting rule. What do you think @olaf-k ?
> 
> my gut feeling would be "yes please!" or at least keep only rules that prevent readability issues (weird capitalization, mix tab/space, this sort of stuff)

aware of any such preceding practice?

**Comment by olaf-k** [comment]
> aware of any such preceding practice?

following our chat: it makes sense to keep some formatting rules to keep the code clean and readable for team members and minimize commit noise, while removing strict rules (e.g., no empty functions) for test files.

**Review by olaf-k — CHANGES_REQUESTED** [review]
Note that the default linting target uses nx's `run-many` to check all projects [and thus skips warnings](https://github.com/nrwl/nx/issues/9638#issuecomment-1118568347).

To get the warnings displayed you must either add `--verbose` to the linting target command or simple run it project by project (e.g., `npx nx lint components`).

At the time of writing, there are still 400+ warning messages from the components project, repeating ones from tsdoc complaining about jsdoc formatting.

**Comment by rinaok** [comment]
> Note that the default linting target uses nx's `run-many` to check all projects [and thus skips warnings](https://github.com/nrwl/nx/issues/9638#issuecomment-1118568347).
> 
> To get the warnings displayed you must either add `--verbose` to the linting target command or simple run it project by project (e.g., `npx nx lint components`).
> 
> At the time of writing, there are still 400+ warning messages from the components project, repeating ones from tsdoc complaining about jsdoc formatting.

@olaf-k good catch!

**Review by olaf-k — APPROVED** [review]
Seeing the diff, there's a bunch of comments that could use some cleanup, but that's beyond the scope of this PR.

**Pull request closed without merging**